### PR TITLE
added an "exists" attribute to differentiate whether we have data or not

### PR DESCRIPTION
**Pull request by christophe-g**
follows discussions #33.

### CODE REVIEW

File: firebase-query.html [hunk 1]
```diff
@@ -331,6 +334,9 @@
             this.async(function() {
               this.syncToMemory(function() {
                 this.splice('data', this.__indexFromKey(key), 1);
+                if (this.data.length === 0) {
```
Comment (Phrohdoh) [inline firebase-query.html]: Doesn't this imply the data in memory does not reflect what is in Firebase?

If that is the case then this isn't true for `[]` which could very well exist in either location and be valid.

Comment (christophe-g) [inline firebase-query.html]: Thanks, 
http://stackoverflow.com/questions/15408416/how-to-handle-empty-arrays-in-firebase : "there is no way to force empty array into firebase".

And yes, data in memory does not always reflect what is in Firebase - see discussion #177 (zeroValue).

**Comment by christophe-g** [comment]
Please note that if you want to and as a temporary measure, you can use my #dev version combining pull request #177, #166 and #167 via 
```
bower install --save  PolymerEl/polymerfire#dev
```

**Comment by mbleigh** [comment]
Would you consider adding a `statusKnown` property to this PR in addition to `exists`? Right now you're having to use a three-value boolean (`null`, `false`, `true`) to track state which feels off to me. Ideally I'd prefer `statusKnown` which indicates whether data has ever been received for the current path, and `exists` which indicates whether it's null.

**Comment by christophe-g** [comment]
Thanks a lot for looking at this and merging #166 and #167. 

Makes perfect sense to add a `statusKnown` attribute. Should land shortly.

**Comment by christophe-g** [comment]
Landed

**Comment by diegopitt** [comment]
Thank you, is there documentation on this implementation? Can you share how exist and statusknown will work?

**Comment by andrewspy** [comment]
Any update on merging `statusKnown` attribute?

**Comment by christophe-g** [comment]
Seems the codebase diverted quite a bit and this PR would need to be rebased against master. I'll do that once https://github.com/firebase/polymerfire/pull/213 is merged. 
C.

**Comment by christophe-g** [comment]
Closed by #237

**Pull request closed without merging**